### ISSUE TITLE
`defmsg` only works for some types of specs

### ISSUE THREAD

**Issue by bhb** [bhb/expound#101]
If you attach a message to say, a set-based spec like

`(expound/def ::myspec #{1 2 3} "should be foo")`

it won't print the message

**Comment by hlship** [comment]
The suggestion, to wrap the set as `#(contains? #{1 2 3} %)` does work.

**Comment by bhb** [comment]
Related: https://github.com/bhb/expound/pull/96 - this would allow users to add messages to existing spec messages (or overwrite them completely)

**Comment by jsa-aerial** [comment]
If this could be made to work, it would be akin to a game changer. But even for something as simple as this:

```Clojure
(s/def ::id (s/or :idint ::id-int :idstg ::id-stg))
(ex/defmsg ::id "Should be either an integer or string of digits")
(ex/expound ::id "abc")
```
You just get the 'standard' error 

Also, trying to use the 'fix' of #96 does not work either:

```Clojure
(ex/def ::id (s/or :idint ::id-int :idstg ::id-stg) "Should be either an integer or string of digits")
(ex/expound ::id "abc")
```
Just results in standard error.

So, should I give up on this?

**Comment by bhb** [comment]
@jsa-aerial Thanks for using Expound! Unfortunately, you're correct: there isn't currently a way to add messages for compound specs like the example you posted above. It's on my radar, but probably won't take priority over some other areas (like spec.alpha2 compatibility) in the immediate future.

**Comment by jsa-aerial** [comment]
OK, in any event, I actually am wanting 'end user' type messages. And that actually doesn't look to be in the scope of Expound anyway. So, may as well close this.

**Comment by bhb** [comment]
That’s correct. Expound is focused on messages for developers. Perhaps Phrase would be a better fit here? https://github.com/alexanderkiel/phrase

**Comment by jsa-aerial** [comment]
Yeah, I was looking at that last night before I replied here. It looks to be 'on the right track", but is totally opaque on how / where / why the dispatch mechanism is tied into your specs.  It works for the simple toy examples given, but does not work at all for even a simple 'or' spec. Or at least it defeated me in getting it to work over the course of many permutations...

**Comment by bhb** [comment]
This has been fixed as of 2c4019fcb48c4eea0fa9dbdc4b36e9bf30f64242
